kotlinlib/src/mill/kotlinlib/KotlinModule.scala

```diff
@@ -134,6 +135,83 @@ trait KotlinModule extends JavaModule { outer =>
     ()
   }
 
+  override def docJar: T[PathRef] = T[PathRef] {
+    T.log.info("docJar task shouldn't be used for Kotlin modules, using dokkaJar instead")
```

Comment:
I wonder if we really need this warning? e.g. for both Java and Scala 2.x and Scala 3.x they all use `docJar`, despite being wildly different backend implementations. Can we just let KotlineModule users use `docJar` as well?

Reply:
Sure, I added Dokka backend for `docJar` directly, without a dedicated `dokkaJar` task (although the example is still called `7-dokkajar` to avoid copying lines from `scalalib`).